### ISSUE TITLE
Linguistic Binding in Diffusion Models: Enhancing Attribute Correspondence through Attention Map Alignment

### ISSUE THREAD

**Issue by geekn0rd** [sharif-ml-lab/visual_compositional_generation_survey#24]
# Task: Read the Paper and Do the following 4 steps

- Highlight the paper systematically (green for important notes - blue for very important notes that worth mentioning in the final survey - red for the sentences that we don’t understand or need further attention in the future) [In Zotero].
- Write the Wikipedia summary of the paper in the GitHub issue (comment in this issue).
- Extract the information in the [excel file](https://docs.google.com/spreadsheets/d/15_sh6TqtUvZIJo1-RZxQtsG0Nvg0vPG_IFKCWiGMjdY/edit?usp=sharing).
- Add the related sentences to the [overleaf file](https://www.overleaf.com/8167246262ccxtrzztjvpz). Note that when you add the sentence, don’t forget to write the reference in the format of ”[ #reference number that is mentioned in the Google Doc - Name of the Author (who added this to the overleaf) ]” at the end of the sentence.

**Comment by geekn0rd** [comment]
## **Introduction and Objective:**
Text-conditioned image generation models often generate incorrect associations between entities and their visual attributes. This reflects a mismatch between linguistic binding of entities and modifiers in the prompt and visual binding of the corresponding elements in the generated image. This issue is caused by the image-text encoder (CLIP) that acts as a bag of words model and fails to capture the syntactic relation between words in a sentence. An approach which first syntactically analyses the prompt to identify entities and their modifiers, and then uses a novel loss function that encourages the cross-attention maps to agree with the linguistic binding reflected by the syntax is proposed by the authors.



## **Methodology: Syntax-Guided Generation**

 - Parse the prompt using spaCy’s transformer-based dependency parser and identify all entity-noun that are not serving as direct modifiers of other nouns. These are the nouns that correspond to objects in the generated image. We then recursively collect all modifiers of the noun into its modifier set ($`\{ S_1, S_2, . . . , S_k \}`$).
 - Consider a pair of a noun and its modifier. We expect the cross-attention map of the modifier to largely overlap with the cross-attention map of the noun, while remaining largely disjoint with the maps corresponding to other nouns and modifiers. To make this happen they designed a loss that operates on all cross-attention maps.

**Loss functions:**

$`L_{pos}(A, S) = \sum_{i} \sum_{(m,n)\in P(S_i)} dist(A_m, A_n)`$

Let $P(S_i)$ represent all pairs (m, n) of tokens between the noun root n and its modifier descendants m in the i-th set $S_i$. The first loss aims to minimize that distance (maximize the overlap) over all pairs of modifiers and their corresponding entity-nouns (m, n).

$`L_{neg} = − \sum_{i} \frac{1}{|U(S_i)|} \sum_{(m,n) \in P(Si)} \sum_{ u \in U(S_i)} dist(A_m, A_u) + dist(A_u, A_n)`$

let $U(S_i)$ represent the set of unmatched words obtained by excluding the words in $S_i$ from the full set of words. The loss encourages moving apart grammatically-unrelated pairs of words.

- Empirically they use the loss of to intervene in the first 25 out of 50 denoising steps. 
- In each of the first 25 steps, a pre-trained denoiser (U-Net) was first used to denoise the latent variable $z_t$. Then, we obtained the cross-attention maps. Next, we used the loss $`L = L_{pos} + L_{neg}`$ to update the latent representation $z_t$ with a gradient step $` z^′_t = z_t − α · ∇_{z_t} L`$. Finally, the U-Net architecture denoises the updated latent variable $z^′_t$ for the next timestep.

<img width="435" alt="Screenshot 2023-11-24 at 4 00 12 in the afternoon" src="https://github.com/sharif-ml-lab/visual_compositional_generation_survey/assets/70889826/ba52cd8f-30be-4c88-9157-afa98205a1c3">



##  **Contributions:**
- A novel method to enrich the diffusion process with syntactic information, using inference-time optimization with a loss over cross-attention maps.
-   A new challenge set of prompts containing a rich number and types of modifiers and entities(DVMP).

###  **Experiments/Results:**
TO DO


###  **Conclusion and Future Work:**
TO DO
